src/sidebar.js

```diff
@@ -1,386 +1,362 @@
-      const touchWidth = this.touchSidebarWidth();
+      const touchWidth = touchSidebarWidth();
 
       if (
-        (this.props.open &&
-          touchWidth <
-            this.state.sidebarWidth - this.props.dragToggleDistance) ||
-        (!this.props.open && touchWidth > this.props.dragToggleDistance)
+        (props.open && touchWidth < sidebarWidth - props.dragToggleDistance) ||
+        (!props.open && touchWidth > props.dragToggleDistance)
       ) {
-        this.props.onSetOpen(!this.props.open);
+        props.onSetOpen(!props.open);
       }
 
-      this.setState({
-        touchIdentifier: null,
-        touchStartX: null,
-        touchCurrentX: null
+      setTouchParams({
+        identifier: null,
+        startX: null,
+        currentX: null
       });
     }
-  }
+  };
 
   // This logic helps us prevents the user from sliding the sidebar horizontally
   // while scrolling the sidebar vertically. When a scroll event comes in, we're
   // cancelling the ongoing gesture if it did not move horizontally much.
-  onScroll() {
-    if (this.isTouching() && this.inCancelDistanceOnScroll()) {
-      this.setState({
-        touchIdentifier: null,
-        touchStartX: null,
-        touchCurrentX: null
+  const onScroll = () => {
+    if (isTouching() && inCancelDistanceOnScroll()) {
+      setTouchParams({
+        identifier: null,
+        startX: null,
+        currentX: null
       });
     }
-  }
+  };
 
   // True if the on going gesture X distance is less than the cancel distance
-  inCancelDistanceOnScroll() {
+  const inCancelDistanceOnScroll = () => {
     let cancelDistanceOnScroll;
 
-    if (this.props.pullRight) {
+    if (props.pullRight) {
       cancelDistanceOnScroll =
-        Math.abs(this.state.touchCurrentX - this.state.touchStartX) <
+        Math.abs(touchParams.currentX - touchParams.startX) <
         CANCEL_DISTANCE_ON_SCROLL;
     } else {
       cancelDistanceOnScroll =
-        Math.abs(this.state.touchStartX - this.state.touchCurrentX) <
+        Math.abs(touchParams.startX - touchParams.currentX) <
         CANCEL_DISTANCE_ON_SCROLL;
     }
     return cancelDistanceOnScroll;
-  }
+  };
 
-  isTouching() {
-    return this.state.touchIdentifier !== null;
-  }
+  const isTouching = () => {
+    return touchParams.identifier !== null;
+  };
 
-  overlayClicked() {
-    if (this.props.open) {
-      this.props.onSetOpen(false);
+  const overlayClicked = () => {
+    if (props.open) {
+      props.onSetOpen(false);
     }
-  }
+  };
 
-  saveSidebarWidth() {
-    const width = this.sidebar.offsetWidth;
+  const saveSidebarWidth = () => {
+    const width = sidebarRef.current.offsetWidth;
 
-    if (width !== this.state.sidebarWidth) {
-      this.setState({ sidebarWidth: width });
+    if (width !== sidebarWidth) {
+      setSidebarWidth(width);
     }
-  }
+  };
 
-  saveSidebarRef(node) {
-    this.sidebar = node;
-  }
+  const saveSidebarRef = node => {
```

Comment:
As we are creating the ref with `useRef` hook I think we don't need this function anymore, what do you think?